hilti/runtime/include/types/stream.h

```diff
@@ -1038,6 +1020,12 @@ class View {
      */
     Offset offset() const { return _begin.offset(); }
 
+    /**
+     * Returns the offset of the view's end location within the associated
+     * stream instance. For an open-ended view, returns the `uint64_t` maximum.
+     */
+    Offset endOffset() const { return _end ? _end->offset() : Offset(std::numeric_limits<uint64_t>::max()); }
```

Comment:
This is great since it allows removing all instances of `unsafeEnd().offset()`.

We already had `Chain::endOffset` in a `detail` namespace with similar signature, but this adds similar code to the public API. With this signature and its magic value when using this function to compute lengths one always needs to check in parallel whether `isOpenEnded` which seems error-prone.

WDYT about instead returning a `std::optional<Offset>` here instead to handle the opened-ended case? If we'd go that we probably should use the same return type for `Stream::offsetEnd` (which would also take care of the documentation not mentioning the magic value there).

Reply:
I'm fine going with optional for the new method. However, a stream does always have an end offset, so don't see a reason to change that.